### PR TITLE
chore: change default thumbnail concurrency and auto-detect container core count

### PR DESCRIPTION
**Pull request by zackpollard**
Drop default thumbnail concurrency to 3 to avoid contention with the default UV_THREADPOOL_SIZE of 4.

Automatically detect container core count, respects docker cgroup limits and then bumps UV_THREADPOOL_SIZE as long as core count is set greater than 4, otherwise it leaves the threadpool size at 4.

### CODE REVIEW

File: server/start.sh [hunk 1]
```diff
@@ -17,4 +17,6 @@ read_file_and_export "DB_USERNAME_FILE" "DB_USERNAME"
 read_file_and_export "DB_PASSWORD_FILE" "DB_PASSWORD"
 read_file_and_export "REDIS_PASSWORD_FILE" "REDIS_PASSWORD"
 
+export UV_THREADPOOL_SIZE=$CPU_CORES
```
Comment (midzelis) [inline server/start.sh:20]: You wouldn't need an export if you just set UV_THREADPOOL_SIZE directly in the .env file.

Comment (zackpollard) [reply server/start.sh:20]: This is a draft PR

**Review by danieldietzler — APPROVED** [review]

File: docs/docs/install/environment-variables.md [hunk 2]
```diff
@@ -38,16 +38,17 @@ Regardless of filesystem, it is not recommended to use a network share for your
...
+| `IMMICH_LOG_LEVEL`              | Log Level (verbose, debug, log, warn, error)    |            `log`             | server, machine learning | api, microservices |
+| `IMMICH_MEDIA_LOCATION`         | Media Location                                  |   `./upload`<sup>\*1</sup>   | server                   | api, microservices |
+| `IMMICH_CONFIG_FILE`            | Path to config file                             |                              | server                   | api, microservices |
+| `IMMICH_WEB_ROOT`               | Path of root index.html                         |      `/usr/src/app/www`      | server                   | api                |
+| `IMMICH_REVERSE_GEOCODING_ROOT` | Path of reverse geocoding dump directory        |     `/usr/src/resources`     | server                   | microservices      |
+| `NO_COLOR`                      | Set to `true` to disable color-coded log output |           `false`            | server, machine learning |                    |
+| `CPU_CORES`                     | Amount of cores available to the immich server  | auto-detected cpu core count | server                   |                    |
```
Comment (bo0tzz) [inline docs/docs/install/environment-variables.md:51]: I just need to complain about how this one line causes the whole table to get reformatted >:(

Comment (zackpollard) [reply docs/docs/install/environment-variables.md:51]: That's markdown for you, the column got wider so it has to reformat everything

File: server/start.sh [hunk 3]
```diff
@@ -17,4 +17,10 @@ read_file_and_export "DB_USERNAME_FILE" "DB_USERNAME"
 read_file_and_export "DB_PASSWORD_FILE" "DB_PASSWORD"
 read_file_and_export "REDIS_PASSWORD_FILE" "REDIS_PASSWORD"
 
+export CPU_CORES="${CPU_CORES:=$(../docker/scripts/get-cpus.sh)}"
```
[the remark this answers is not in the document]
Comment (zackpollard) [reply server/start.sh]: Yea I fixed that, was like that for testing :laughing:

**Review by bo0tzz — APPROVED** [review]

**Pull request merged**